tools/regenerate-docs.ts

```diff
@@ -89,6 +99,36 @@ const updateRulesList = (
   ].join('\n');
 };
 
+/**
+ * Generate a rule doc header for a particular rule.
+ * @param description - rule description
+ * @param name - rule name
+ * @returns {string[]} - lines for new header including marker
+ */
+const generateRuleHeaderLines = (description, name) => {
+  const lines = [`# ${description} (\`${name}\`)`];
+
+  lines.push(...getRuleNoticeLines(name));
+
+  lines.push(END_RULE_HEADER_MARKER);
+
+  return lines;
+};
+
+/**
+ * Replace the header of a doc up to and including the specified marker.
+ * Insert at beginning if header doesn't exist.
+ * @param lines - lines of doc
+ * @param newHeaderLines - lines of new header including marker
+ * @param marker - marker to indicate end of header
+ */
+const replaceOrCreateHeader = (lines, newHeaderLines, marker) => {
```

Comment:
It looks like maybe our typechecking is not running properly on these files - I'll look into that later, but you need to have explicit types on these params as currently they're implicitly `any`.

(You can make typechecking work by adding `tools/**/*` to the `include` array in `tsconfig.json`, but I need to check if that will impact the compiled output before actually committing that)

Reply:
Good catch, added types and updated tsconfig.json. We have `"noEmit": true` so I don't think this will cause any output to be produced?

Reply:
Thanks, but I'd really like to avoid changing the `tsconfig.json` in anyway in this PR just to be on the safe side - its also an indicator that I should do a look over the whole codebase for a bit of a tidy up

Reply:
Okay removed change to `tsconfig.json`.